src/nbl/ext/ImGui/ImGui.cpp

```diff
@@ -23,61 +20,199 @@ using namespace nbl::ui;
+			auto compiler = make_smart_refctd_ptr<CHLSLCompiler>(smart_refctd_ptr(system));																				//! note we are out of default logical device's compiler set scope so also a few special steps are required to compile our extension shaders to SPIRV
+			auto includeFinder = make_smart_refctd_ptr<IShaderCompiler::CIncludeFinder>(smart_refctd_ptr(system));
+			auto includeLoader = includeFinder->getDefaultFileSystemLoader();
+			includeFinder->addSearchPath(NBL_ARCHIVE_ALIAS.data(), includeLoader);
+
+			auto createShader = [&]<core::StringLiteral key, asset::IShader::E_SHADER_STAGE stage>() -> core::smart_refctd_ptr<video::IGPUShader>
 			{
-				const system::SBuiltinFile vertex = ::ext::imgui::spirv::builtin::get_resource<"ext/imgui/spirv/vertex.spv">();
-				const system::SBuiltinFile fragment = ::ext::imgui::spirv::builtin::get_resource<"ext/imgui/spirv/fragment.spv">();
-			} spirv;
+		asset::IAssetLoader::SAssetLoadParams params = {};
+				params.logger = m_creationParams.utilities->getLogger();
+				params.workingDirectory = NBL_ARCHIVE_ALIAS.data();
 
-			auto createShader = [&](const system::SBuiltinFile& in, asset::IShader::E_SHADER_STAGE stage) -> core::smart_refctd_ptr<video::IGPUShader>
-			{
-				const auto buffer = core::make_smart_refctd_ptr<asset::CCustomAllocatorCPUBuffer<core::null_allocator<uint8_t>, true> >(in.size, /*this cast is awful but my custom buffer won't free it's memory*/ (void*)in.contents, core::adopt_memory); // no copy
-				const auto shader = make_smart_refctd_ptr<ICPUShader>(core::smart_refctd_ptr(buffer), stage, IShader::E_CONTENT_TYPE::ECT_SPIRV, "");
-				
-				return m_device->createShader(shader.get());
+				auto bundle = m_creationParams.assetManager->getAsset(key.value, params);
+				const auto assets = bundle.getContents();
+
+				if (assets.empty())
+				{
+					m_creationParams.utilities->getLogger()->log("Could not load \"%s\" shader!", system::ILogger::ELL_ERROR, key.value);
+					return nullptr;
+				}
+
+				const auto shader = IAsset::castDown<asset::ICPUShader>(assets[0]);
+
+				CHLSLCompiler::SOptions options = {};
+				options.stage = stage;
+				options.preprocessorOptions.sourceIdentifier = key.value;
+				options.preprocessorOptions.logger = m_creationParams.utilities->getLogger();
+				options.preprocessorOptions.includeFinder = includeFinder.get();
+
+				auto compileToSPIRV = [&]() -> core::smart_refctd_ptr<ICPUShader>
+				{
+					#define NBL_DEFAULT_OPTIONS "-spirv", "-Zpr", "-enable-16bit-types", "-fvk-use-scalar-layout", "-Wno-c++11-extensions", "-Wno-c++1z-extensions", "-Wno-c++14-extensions", "-Wno-gnu-static-float-init", "-fspv-target-env=vulkan1.3", "-HV", "202x" /* default required params, just to not throw warnings */
```

Comment:
a better solution is to pass a custom logger to the compile command and silence that warning via a regexpr filter instead

the default compile options will change in the future, so be more maintanble

Reply:
> a better solution is to pass a custom logger to the compile command and silence that warning via a regexpr filter instead
> 
> the default compile options will change in the future, so be more maintanble

I would use the getter to sync with current defaults, but someone made it return wchars array xD

Reply:
yep, I think I will better "polish" the getter with an overload or something + use it to sync

Reply:
ok do that then

Reply:
using args directly from the class, had to provide a few updates anyway